jablib/src/main/java/org/jabref/logic/ocr/OcrService.java

```diff
@@ -0,0 +1,78 @@
+package org.jabref.logic.ocr;
+
+import net.sourceforge.tess4j.Tesseract;
+import net.sourceforge.tess4j.TesseractException;
+import org.jabref.model.strings.StringUtil;  // JabRef utility class
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.nio.file.Path;
+
+/**
+ * Service for performing Optical Character Recognition (OCR) on PDF files.
+ * This class provides a high-level interface to OCR functionality,
+ * abstracting away the specific OCR engine implementation details.
+ */
+public class OcrService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(OcrService.class);
+
+    // The OCR engine instance
+    private final Tesseract tesseract;
+
+    /**
+     * Constructs a new OcrService with default settings.
+     * Currently uses Tesseract with English language support.
+     */
+    public OcrService() {
+        this.tesseract = new Tesseract();
+
+        // Configure Tesseract
+        tesseract.setLanguage("eng");
+
+        // TODO: This path needs to be configurable and bundled properly
+        // For now, we'll use a relative path that works during development
+        tesseract.setDatapath("tessdata");
+
+        LOGGER.debug("Initialized OcrService with Tesseract");
+    }
+
+    /**
+     * Performs OCR on a PDF file and returns the extracted text.
+     *
+     * @param pdfPath Path to the PDF file to process
+     * @return The extracted text, or empty string if no text found
+     * @throws OcrException if OCR processing fails
+     */
+    public String performOcr(Path pdfPath) throws OcrException {
+        // Validate input
+        if (pdfPath == null) {
+            throw new OcrException("PDF path cannot be null");
+        }
+
+        File pdfFile = pdfPath.toFile();
+        if (!pdfFile.exists()) {
+            throw new OcrException("PDF file does not exist: " + pdfPath);
+        }
```

Comment:
Use exceptions only for exceptional states. See https://github.com/HugoMatilla/Effective-JAVA-Summary?tab=readme-ov-file#57-use-exceptions-only-for-exceptional-conditions